client/components/Buttons/LikeButton.vue

```diff
@@ -23,39 +23,22 @@ export default Vue.extend({
   },
   data() {
     return {
-      isFavorite: false,
-      // eslint-disable-next-line @typescript-eslint/no-empty-function
-      unsubscribe(): void {}
+      isFavorite: false
     };
   },
-  watch: {
-    item: {
-      immediate: true,
-      handler(): void {
-        this.isFavorite = this.item.UserData?.IsFavorite || false;
-      }
+  computed: {
+    ...mapGetters('items', ['getItem']),
+    observedItem(): BaseItemDto {
+      return this.getItem(this.item.Id as string);
     }
   },
-  activated() {
-    this.unsubscribe = this.$store.subscribe((mutation) => {
-      if (
-        mutation?.type === 'SOCKET_ONMESSAGE' &&
-        mutation?.payload?.MessageType === 'UserDataChanged'
-      ) {
-        const payloadData = mutation?.payload?.Data?.UserDataList;
-
-        if (payloadData) {
-          for (const payloadItem of payloadData) {
-            if (payloadItem.ItemId === this.item.Id) {
-              this.isFavorite = payloadItem.IsFavorite;
-            }
-          }
-        }
-      }
-    });
+  watch: {
+    observedItem(): void {
+      this.isFavorite = this.observedItem?.UserData?.IsFavorite || false;
+    }
   },
-  deactivated() {
-    this.unsubscribe();
+  async created() {
+    await this.$userLibrary.getItem(this.item.Id as string);
```

Comment:
This forces fetching the item, so it kinda defeats the purpose of the store. Maybe check for its presence first.

Reply:
I think it's there because most of the places don't use the store at the moment. We should finish implementing the store everywhere before refactoring this component, imo.

Reply:
Then I'd say just check for presence and fetch if needed :p

Reply:
Ideally, they'd be fetched outside of this, though, hence why I think doing all the rest before refactoring this is the better route :yum:

Reply:
I agree. But I think it doesn't cost too much to add the if for now :)